### PR TITLE
fix(carousel): carousel scroll behaviour when interacted with cta

### LINKED ISSUE #622: Fix the advertising Carousel scroll behaviour when CTA is interacted with or has focus
If a CTA on one of the advertising tiles has focus, and the animation / scrolling is on, it breaks.

This will be fixed in Canopy.

### Tabbing between the controls on the ads
Legal   General - Online Self Service

### Clicking to an overlay and then coming back

I assume the issue here is that the ad has scrolled to the next ad when focus is returned (overlays / modals must return focus to the button that opened them when the overlay is closed [for accessibility]).
from-overlay

For the first issue, perhaps the solution is to disable the animation if an ad has focus? I'm not sure what we can do about the second issue as focus will have gone to the overlay.

Related issue in OSS: [DCI-17572](https://landg.spectrumhosting.net/jira/browse/DCI-17572)

### PR DESCRIPTION
**Pull request by Nupur04**
# Description
Stop carousel when any button inside carousel item has been clicked and fix broken view on tab
switch

If a CTA on one of the advertising tiles has focus, and the animation / scrolling is on, it breaks.
Tabbing between the controls on the ads DCI-17572

Fixes #622 

# Checklist:

- [x] The commit messages follow the [convention for this project](./blob/master/docs/CONTRIBUTING.md#conventional-commits)
- [x] I have provided an adequate amount of test coverage
- [x] I have added the functionality to the [test app](./blob/master/docs/CONTRIBUTING.md#build-test-application)
- [x] I have provided a story in storybook to document the changes
- [x] I have provided documentation in the notes section of the story
- [x] I have added any new public feature modules to public-api.ts

### CODE REVIEW

File: projects/canopy/src/lib/carousel/carousel.component.ts [hunk 1]
```diff
@@ -41,6 +41,14 @@ export class LgCarouselComponent implements AfterContentInit, OnDestroy {
   pause = new BehaviorSubject<boolean>(false);
   pausableTimer$: Observable<void>;
 
+  pauseCarousel() {
```
Comment (elenagarrone) [inline projects/canopy/src/lib/carousel/carousel.component.ts:44]: Could you please add a return type of `void` to these two new functions?

Comment (Nupur04) [reply projects/canopy/src/lib/carousel/carousel.component.ts:44]: sure

**Comment by elenagarrone** [comment]
@Nupur04 would you mind updating the commit message context from `carousel-focus` to `carousel` please?

File: projects/canopy/src/lib/carousel/carousel.component.ts [hunk 2]
```diff
@@ -41,6 +41,14 @@ export class LgCarouselComponent implements AfterContentInit, OnDestroy {
...
+    this.pause.next(true);
+  }
+
+  playCarousel() {
```
Comment (elenagarrone) [inline projects/canopy/src/lib/carousel/carousel.component.ts:48]: Where are you using these functions? are they not supposed to be linked to the template?

Comment (Nupur04) [reply projects/canopy/src/lib/carousel/carousel.component.ts:48]: Yes, they are in template. I think I resolved the conflict and methods got removed. Will update the PR.

Comment (Nupur04) [reply projects/canopy/src/lib/carousel/carousel.component.ts]: These 2 methods Play and Pause carousel will be exposed by targeting the carousel with ViewChild decorators in the application(OSS) to fix the accessibility issues.

File: projects/canopy/src/lib/carousel/carousel.stories.ts [hunk 3]
```diff
@@ -28,14 +28,14 @@ const carouselItems = `
   <p>Carousel items scale vertically to the largest item in the carousel and scale horizontally to the width of the parent container.</p>
   <p>Carousels are built to work with varied content sizes but will work best if the content of each carousel item is of a similar size</p>
   <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Odit maiores, dolor suscipit ullam omnis corrupti laboriosam, amet, rem fugiat id deleniti ipsum possimus. At assumenda, asperiores dolorem temporibus quaerat libero quos blanditiis quod in? Magni laborum tempora, laudantium facilis ut sunt totam beatae repellat doloribus blanditiis sed repellendus maiores necessitatibus iste ex quod voluptate? Nesciunt, libero officiis labore molestias enim doloribus id assumenda dicta numquam fuga velit reiciendis quae praesentium soluta eaque exercitationem? Ullam iste culpa accusantium tenetur repellendus sit sunt error nisi dicta deleniti cum amet, voluptates soluta et laudantium quis.</p>
-</lg-carousel-item>
+  </lg-carousel-item>
```
Comment (dannyoz) [inline projects/canopy/src/lib/carousel/carousel.stories.ts:31]: Such a minor thing but can you revert the additional white space added to the template file?

**Review by dannyoz — APPROVED** [review]

**Review by sacalusalina — DISMISSED** [review]

File: projects/canopy/src/lib/carousel/carousel.notes.ts [hunk 4]
```diff
@@ -41,4 +41,6 @@ and in your HTML:
 | \`\`headingLevel\`\`  | The heading level of the description: \`\`1\`\`, \`\`2\`\`, \`\`3\`\`, \`\`4\`\`, \`\`5\`\`, \`\`6\`\`   | number  | undefined |   Yes    |
 | \`\`autoPlay\`\`      | Enables auto play mode when set to true.                                                                 | boolean |   false   |   No     |
 | \`\`autoPlayDelay\`\` | Delay time in milliseconds to switch to next slide when autoPlay mode is set to true.                    | number  |   2000    |   No     |
+
+## PlayCarousel and PauseCarousel methods can be exposed by targeting the Carousel with ViewChild decorators to fix the accessibility issues.
```
Comment (elenagarrone) [inline projects/canopy/src/lib/carousel/carousel.notes.ts:45]: Not sure you have to specify `to fix the accessibility issues` as most people wouldn't know what you are referring to 🤔

Comment (Nupur04) [reply projects/canopy/src/lib/carousel/carousel.notes.ts:45]: Good point.

**Review by elenagarrone — APPROVED** [review]

**Review by sacalusalina — APPROVED** [review]

**Review by dannyoz — APPROVED** [review]

**Review by owensgit — APPROVED** [review]

**Review by elenagarrone — APPROVED** [review]

**Review by owensgit — APPROVED** [review]

**Pull request merged**